middle_end/flambda2/to_cmm/to_cmm_primitive.ml

```diff
@@ -297,6 +349,12 @@ let array_load ~dbg (array_kind : P.Array_kind.t)
       Debuginfo.print_compact dbg
   | Values, Naked_vec128s ->
     Misc.fatal_error "Attempted to load a SIMD vector from a value array."
+  | Unboxed_product _, Naked_vec128s ->
+    (* CR mshinwell: should this be supported? *)
```

Comment:
Seems fine to support, just need to make sure the element num computation accounts for vectors taking up >1 word